Trades/Trades.go

```diff
@@ -14,39 +14,71 @@ type TradeItems struct {
 }
 
 // Trade function that
-func Trade(TradeObject, TradeObjectBuffer []string) []byte {
+func Trade(tradeObjects, tradeObjectsBuffer []string) (tradeObjectsProcessed, tradeObjectsBufferProcessed []string,
+	TradedItemJSON []byte) {
+
+	//If the TradeObject slice is empty (because all previously moved to buffer) - empty buffer back to main slice
+	tradeObjects, tradeObjectsBuffer = emptyBufferTradesToMain(tradeObjects, tradeObjectsBuffer)
+
+	//Create a 'random' trade
+	TradedItem, randomObjNumber := randomTrade(tradeObjects)
+
+	//Copy traded item to buffer
+	tradeObjects, tradeObjectsBuffer = moveTradeObjectToBuffer(tradeObjects, tradeObjectsBuffer, randomObjNumber)
+
+	//Remove the traded item from the 'TradeObjects' so duplicates don't occur until all items have been traded at least once
+	tradeObjects = removeTradedObject(tradeObjects, randomObjNumber)
+
+	//Marshall the traded struct into JSON format
+	TradedItemJSON, err := json.Marshal(TradedItem)
+	if err != nil {
+		fmt.Println(err)
```

Comment:
in case of an error, `TradedItemJSON` is undefined, and you don't want to proceed executing the code after this. hence we return the error to the caller.

this means, we also must change our return to return the error alongside the data that the function is expected to return.